dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+            RequestWrapper rw = new RequestWrapper( request );
+            if ( request.getAttribute( "User-Agent" ) != null && request.getAttribute( "User-Agent" ).equals( Constants.USER_AGENT_DOTCMS_BROWSER ) ) {
+                rw.setCustomUserAgentHeader( Constants.USER_AGENT_DOTCMS_BROWSER );
+            }
+            context = new ChainedContext( ctx, getEngine(), rw, response, Config.CONTEXT );
+        }
+
+        context.put("context", context);
+		Logger.debug(VelocityUtil.class, "ChainedContext=" + context);
+		/*
+		 * if we have a toolbox manager, get a toolbox from it See
+		 * /WEB-INF/toolbox.xml
+		 */
+		context.setToolbox(getToolboxManager().getToolboxContext(context));
+
+
+		// put the list of languages on the page
+		context.put("languages", getLanguages());
+		HttpSession session = request.getSession(false);
+		if(!UtilMethods.isSet(request.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)) && session!=null)
+		    context.put("language", (String) session.getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE));
+		else
+		    context.put("language", request.getAttribute(WebKeys.HTMLPAGE_LANGUAGE));
+
+		try {
+			Host host;
+			host = WebAPILocator.getHostWebAPI().getCurrentHost(request);
+			context.put("host", host);
+		} catch (Exception e) {
+			Logger.error(VelocityUtil.class,e.getMessage(),e);
+		}
+		context.put("pdfExport", false);
+        context.put("dotPageMode", PageMode.get(request));
+		if(request.getSession(false)!=null){
+			try {
+				User user = (com.liferay.portal.model.User) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.CMS_USER);
+				context.put("user", user);
+
+				Visitor visitor = (Visitor) request.getSession().getAttribute(WebKeys.VISITOR);
+				context.put("visitor", visitor);
+
+			} catch (Exception nsue) {
+				Logger.error(VelocityUtil.class, nsue.getMessage(), nsue);
+			}
+		}
+		return context;
+
+	}
+
+	public static String mergeTemplate(String templatePath, Context ctx) throws ResourceNotFoundException, ParseErrorException, Exception{
+		VelocityEngine ve = VelocityUtil.getEngine();
+		Template template = null;
+		StringWriter sw = new StringWriter();
+		template = ve.getTemplate(templatePath);
+
+		template.merge(ctx, sw);
+
+		return sw.toString();
+		
+	}
+	
+	public static String eval(String velocity, Context ctx) throws ResourceNotFoundException, ParseErrorException, Exception{
+		VelocityEngine ve = VelocityUtil.getEngine();
+		StringWriter sw = new StringWriter();
+		ve.evaluate(ctx, sw, "velocity eval", velocity);
+		return sw.toString();
+		
+	}
+	private static ToolboxManager toolboxManager=null;
+    public static ToolboxManager getToolboxManager () {
+        if ( toolboxManager == null ) {
+            synchronized ( VelocityUtil.class ) {
+                if ( toolboxManager == null ) {
+                    toolboxManager = ServletToolboxManager.getInstance( Config.CONTEXT, Config.getStringProperty("TOOLBOX_MANAGER_PATH", "/WEB-INF/toolbox.xml"));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 162 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)